src/Execute/ExecutionEnvManager.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {execSync} from 'child_process';
+import * as vscode from 'vscode';
+
+import {Command} from '../Backend/Command';
+import {Executor} from '../Backend/Executor';
+import {Balloon} from '../Utils/Balloon';
+
+import * as Envs from './Envs/Envs';
+import {ExecutionEnv} from './ExecutionEnv';
+
+// This is a concept for connected device, which means local or remote.
+// TODO Handle about remote case.
+// Currently, only local case Handled.
+
+interface ExectionEnvEnvManager {
+  runInference(envName: string, executor: Executor, modelPath: string): void;
+  runProfile(envName: string, executor: Executor, modelPath: string): void;
+  refreshExecutionEnv(): void;
+}
+
+class BaseEnvManager implements ExectionEnvEnvManager {
+  executionEnvs: Map<string, ExecutionEnv>;
+  constructor() {
+    this.executionEnvs = new Map<string, ExecutionEnv>();
+  }
+  runInference(envName: string, executor: Executor, modelPath: string): void {
+    throw new Error('Method not implemented.');
+  }
+  runProfile(envName: string, executor: Executor, modelPath: string): void {
+    throw new Error('Method not implemented.');
+  }
+  refreshExecutionEnv(): void {
+    throw new Error('Method not implemented.');
+  }
+}
+
+interface ExecutionEnvManagerMap {
+  [key: string]: BaseEnvManager;
+}
+
+let globalManagerMap: ExecutionEnvManagerMap = {};
+// This list will keep envTypes on it.
+let envList = Object.values(Envs);
```

Comment:
Some backend provider may want to implement, e.g., `RaspberryExecutionEnv` in their backend.
In such case `envList` won't contain `RaspberryExecutionEnv`.
I wonder if such situation is OK or not.

Reply:
I thought anyone could create new EnvType on this, and every time they create new type, they need to add codes on `EnvManager` constructor like below

```
for device in RaspberryExecutionEnv.DeviceList():
    addDeviceList(new RaspberryExecutionEnv(device))
``` 

so, I extract all `ExecutionEnv` on `Env/Envs.ts` to export, and use from `Envs.ts` module.

Reply:
as talked on offline with @hyunsik-yoon , if Backend Provider want to hide there own `ExecutionEnv`, this example could not cover this. so, I think we need to add a method to add this on private backend.